pkg/miscellaneous/miscellaneous.go

```diff
@@ -296,11 +297,11 @@ func GetEnvVarOrError(varName string) (string, error) {
 // GetEtcdSvcEndpoint returns the endpoint to the etcd client service
 func GetEtcdSvcEndpoint() (string, error) {
 	var inputFileName string
-	etcdConfigForTest := os.Getenv("ETCD_CONF")
-	if etcdConfigForTest != "" {
+
+	inputFileName = GetConfigFilePath()
+	if inputFileName != EtcdConfigFilePath {
```

Comment:
Can you explain why this check is required or was not there before?

Reply:
So ideally if we are in test-mode (ETCD_CONF is set) then we return an empty service endpoint so that the default one is used (`http://127.0.0.1:2379`)
We send a proper service endpoint only when running in a cluster

Reply:
Hm, I can't really judge this case but it seems very implicit and also confusing (because why should we then supprt setting a config via `ETCD_CONF` ?!). At a minimum a comment would be nice why such a check is performed.

Reply:
Yes, I understand, and we will be picking this up and finding a better way to implement this while refactoring the code base
For the time being I have added a comment here